CartesianCategories/gap/SymmetricCartesianClosedCategoriesDerivedMethods.gi

```diff
@@ -879,7 +879,7 @@ AddDerivationToCAP( CartesianPostComposeMorphismWithGivenObjects,
     
     braiding := CartesianBraiding( cat, ExponentialOnObjects( cat, b, c ), ExponentialOnObjects( cat, a, b ) );
     
-    return PreCompose( cat, braiding, CartesianPreComposeMorphism( cat, a, b, c ) );
+    return PreComposeList( cat, source, [ braiding, CartesianPreComposeMorphism( cat, a, b, c ) ], range );
```

Comment:
I don't think this is the desired solution. The source of `PreCompose` is the source of the first morphism, i.e. the source of `braiding`. So the desired solution is that `braiding` has the expected source in the first place.

Thinking more about this, I think we should never use `PreComposeList` for literal lists. The information about source and range is already available in this case, so adding source and range explicitly does not add anything if everything else is correct. And if things are wrong, adding source and range only hides actual errors. For example, if `braiding` would have the wrong source here, using PreComposeList would just hide the wrong source from CompilerForCAP.